### ISSUE TITLE
Use stable k8s api versions for resource which have been promoted to GA

### ISSUE THREAD

**Issue by den-is** [linkerd/linkerd2#2960]
### Prelude
As of STEP0 in your getting started guide you officially request users:
> Step 0: Setup
> Before we can do anything, we need to ensure you have access to a Kubernetes cluster running 1.10.0 or later

### What problem are you trying to solve?
But at the same time `linkerd` generates:
```yaml
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: linkerd-identity
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: linkerd-controller
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: linkerd-web
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: linkerd-prometheus
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: linkerd-grafana
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: linkerd-sp-validator
---
kind: DaemonSet
apiVersion: extensions/v1beta1
metadata:
  name: linkerd-cni
---
kind: ClusterRoles
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: linkerd-linkerd-identity
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: linkerd-linkerd-identity
---
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: linkerd-linkerd-controller
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: linkerd-linkerd-controller
---
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: linkerd-linkerd-prometheus
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: linkerd-linkerd-prometheus
---
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1 # correct
metadata:
  name: linkerd-linkerd-sp-validator
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1 # correct
metadata:
  name: linkerd-linkerd-sp-validator
```
### How should the problem be solved?
DaemonSet, Deployment, ReplicaSet - are stable since kubernetes **v1.9** and are in **apps/v1**
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md#apps

rbac.authorization.k8s.io API group is GA and v1 since kubernetes **v1.8**
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.8.md#sig-auth

So I almost always I inspect/update third party vendor manifests before running them.
It will be very nice to have these resources updated to a normal versions.

**Comment by Pothulapati** [comment]
Also, on this issue `deployments`,etc Resource types from `extensions/v1beta1` will be depreciated in Kubernetes 1.16. So, Its important that we move to `apps/v1beta1`

https://groups.google.com/forum/#!topic/kubernetes-dev/je0rjyfTVyc/discussion

**Comment by den-is** [comment]
why `apps/v1beta1` when there is `apps/v1` and 99% of other project have moved to these APIs, even most established.
